icechunk-python/python/icechunk/_icechunk_python.pyi

```diff
@@ -323,4 +325,132 @@ class StoreConfig:
+    """Enum for selecting the which version of a conflict"""
+
+    Fail = 0
+    UseOurs = 1
+    UseTheirs = 2
+
+class ConflictSolver:
+    """An abstract conflict solver that can be used to detect or resolve conflicts between two stores
+
+    This should never be used directly, but should be subclassed to provide specific conflict resolution behavior
+    """
+
+    ...
+
+class BasicConflictSolver(ConflictSolver):
+    """A basic conflict solver that allows for simple configuration of resolution behavior"""
+
+    def __init__(
+        self,
+        *,
+        on_user_attributes_conflict: VersionSelection = VersionSelection.UseOurs,
+        on_chunk_conflict: VersionSelection = VersionSelection.UseOurs,
+        fail_on_delete_of_updated_array: bool = False,
+        fail_on_delete_of_updated_group: bool = False,
+    ) -> None:
+        """Create a BasicConflictSolver object with the given configuration options
+        Parameters:
+        on_user_attributes_conflict: VersionSelection
+            The behavior to use when a user attribute conflict is encountered, by default VersionSelection.use_ours()
+        on_chunk_conflict: VersionSelection
+            The behavior to use when a chunk conflict is encountered, by default VersionSelection.use_theirs()
+        fail_on_delete_of_updated_array: bool
+            Whether to fail when a chunk is deleted that has been updated, by default False
+        fail_on_delete_of_updated_group: bool
+            Whether to fail when a group is deleted that has been updated, by default False
+        """
+        ...
+
+class ConflictDetector(ConflictSolver):
+    """A conflict detector that can be used to detect conflicts between two stores and
+    report if resolution is possible
+    """
+
+    def __init__(self) -> None: ...
+
+class IcechunkError(Exception):
+    """Base class for all Icechunk errors"""
+
+    ...
+
+class ConflictErrorData:
+    """Data class for conflict errors. This describes the snapshot conflict detected when committing a session
+
+    If this error is raised, it means the branch was modified and committed by another session after the session was created.
+    """
+    @property
+    def expected_parent(self) -> str:
+        """The expected parent snapshot ID.
+
+        This is the snapshot ID that the session was based on when the
+        commit operation was called.
+        """
+        ...
+    @property
+    def actual_parent(self) -> str:
+        """
+        The actual parent snapshot ID of the branch that the session attempted to commit to.
+
+        When the session is based on a branch, this is the snapshot ID of the branch tip. If this
+        error is raised, it means the branch was modified and committed by another session after
+        the session was created.
+        """
+        ...
+
+class PyConflictError(IcechunkError):
+    """An error that occurs when a conflict is detected"""
+
+    args: tuple[ConflictErrorData]
```

Comment:
Why is this just a one-item tuple? Wouldn't 
```suggestion
    conflicts: ConflictErrorData
```

be simpler?

Reply:
This wrapped in a python exception now, so this is hidden from UX